zookeeper-contrib/zookeeper-contrib-rest/src/main/java/org/apache/zookeeper/server/jersey/RestMain.java

```diff
@@ -53,7 +53,7 @@ public void start() throws IOException {
        System.out.println("Starting grizzly ...");
 
        boolean useSSL = cfg.useSSL();
-       gws = new GrizzlyWebServer(cfg.getPort(), "/tmp/23cxv45345/2131xc2/", useSSL);
+       gws = new GrizzlyWebServer(cfg.getPort(), System.getProperty("java.io.tmpdir","/tmp") + "/23cxv45345/2131xc2/", useSSL);
```

Comment:
Can we use this utility function ?
https://docs.oracle.com/javase/7/docs/api/java/nio/file/Files.html#createTempDirectory%28java.nio.file.Path,%20java.lang.String,%20java.nio.file.attribute.FileAttribute...%29

Reply:
web resource path has to be a static path. createTempDirectory will make it dynamic which will break the functionality

Reply:
yes, seems functionality will change if directory is created dynamically. How someone will put web resources in that dynamically created directory. So lets not use createTempDirectory.

What you say @eolivelli

Reply:
Then I don't understand why we want to use tmpdir. It contains ephemeral files, that won't be present at the next execution of the program. 

I thought we just had to set some existing directory, without any particular meaning

Reply:
` It contains ephemeral files, that won't be present at the next execution of the program.`
Makes sense to use dynamically created files

Reply:
Sp why can't we use the official method to create an unique temporary directory?

Reply:
it can be used.

Reply:
so I believe it is better to use `createTempDirectory`
the '/23cxv45345/2131xc2/' suffix is useless